subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/ResolutionResultApiIntegrationTest.groovy

```diff
@@ -80,4 +81,132 @@ bar:1.0 requested
 baz:1.0 requested
 """
     }
+
+    def "resolution result API gives access to dependency reasons in case of conflict"() {
```

Comment:
Hmmm. This seems to be the test coverage I was looking for, but I was searching for `byReason` from the `ResolveTestFixture`. Seems like this test would be simpler if it leveraged the fixture.

Reply:
In fact, this is coverage for the complex case (conflict resolution), but we don't have coverage for the basic case of a simple dependency-with-reason showing up in the resolution result. (The previous test does not provide custom reasons)